proposals/0357-alpenglow_validator_admission_ticket.md

```diff
@@ -0,0 +1,219 @@
+
+- a BLS public key
+
+- at least 1.6 SOL VAT fee plus the necessary storage rent amount for a new
+epoch in its corresponding identity account
+
+When the staked validators for a new epoch are calculated, the leader will
+perform the following operations:
+
+- When there are more than 2,000 valid validators, sort all valid vote accounts
+by descending order of stake. If some validator with stake S is in position
+2001, then we remove all validators with stake S and less. If there are fewer
+than 2,000 valid validators, pick all of them.
+
+- Deduct 1.6 SOL VAT fee from each picked vote identity account once
+
+- Mark the fee burned and write the result into the bank
+
+All validators must perform the same operation, or the resulting bankhash will
+be different.
+
+### How to implement the checks
+
+1. When a new bank crosses an epoch boundary (bank.epoch() >
+parent_bank.epoch()), calculate the participating staked validators for the
+next epoch (bank.epoch() + 1). This is the same as now.
+
+2. Perform stake activation and deactivation so that the intended stake values
+are used for the new epoch.
+
+3. The calculation goes through all vote accounts and filters those that meet
+the following criteria:
+
+  - The corresponding identity account has a balance of at least 1.6 SOL
+
+  - The account has a valid BLS compressed public key (i.e., it can be
+    correctly decompressed)
+
+4. If the number of filtered accounts exceeds 2,000, sort them according to the
+following rules and select the top 2,000. Otherwise, return the entire list:
+
+  - Sort in descending order of stake (largest to smallest)
+
+  - If multiple validators have exactly the same amount of stake and including
+  all of them would exceed the 2,000 limit, then exclude all of them
+
+5. Subtract 1.6 SOL from the corresponding vote identity account for each
+validator in the accepted list from the previous step
+
+6. Record the VAT fee subtraction in the bank, which reduces the bank’s
+capitalization. This occurs before any transactions are processed in the new
+epoch’s bank
+
+7. The epoch stakes for epoch bank.epoch() + 1 are generated from the list
+produced in step 4. These epoch stakes are saved in bank snapshots, so if the
+network restarts from a snapshot taken from any bank within the same epoch as
+bank.epoch(), the 1.6 SOL fee will not be charged again
+
+## Operation Considerations
+
+- To be included in epoch e+1, validator operators must ensure the vote
+identity account has at least 1.6 SOL before epoch e-1 ends
+
+- Validator operators must ensure they have valid BLS public key specified in
+their vote account
+
+  - The ledger tools will be updated to verify validity of the BLS public key
+when performing updates
+
+  - We will not allow removing the BLS public key
+
+  - The written BLS public key must be generated by the vote authority ed25519
+keypair and change with the vote authority keypair, the keygen tools and
+ledger tools will be updated
```

Comment:
these are relevant to whatever simd specifies the vote program instruction for setting the bls key, not this one

Reply:
Removed.